src/main/java/com/dougnoel/sentinel/system/FileManager.java

```diff
@@ -155,6 +167,30 @@ public static BufferedImage readImage(String subDirectory, String fileName) {
 	* @return String the constructed path as a String
 	*/
 	private static String getImagePath(String subDirectory, String fileName) {
-		return Configuration.toString("imageDirectory", "logs/images").replace("/", File.separator) + File.separator + subDirectory + File.separator + fileName;
+		String outputSubDir = "";
+
+		if(subDirectory != null){
+			outputSubDir = subDirectory;
+		}
+		return Configuration.toString("imageDirectory", "logs/images").replace("/", File.separator) + File.separator + outputSubDir + File.separator + fileName;
+	}
+
+	/**
+	 * Verifies a given file exists within a configured timeout
+	 *
+	 * @param fileToCheck File the file object of the file we wish to verify exists
+	 * @throws IOException in the case the file does not begin existing within the default timeout
+	 */
+	private static void verifyFileCreation(File fileToCheck) throws IOException {
+		var errorMessage = SentinelStringUtils.format("Failed to create the {} image file", fileToCheck.getAbsolutePath());
+
+		long searchTime = Time.out().getSeconds() * 1000;
+		long startTime = System.currentTimeMillis(); // fetch starting time
+		while ((System.currentTimeMillis() - startTime) < searchTime) {
+			if (Files.exists(fileToCheck.toPath()))
```

Comment:
Was this taking a long time to search? It feels odd to need a timeout on something like this.

Reply:
It was not. The Time.out().getSeconds(), remember, just gets the number of seconds before aborting, not an actual pause. This'll operate as fast as it needs to.

It may also no longer be necessary, but I did see a failure due to file creation (It may have been a mad location at the time, though). If you want we can pair up and go over it to see if we can trim it out.